client/src/components/chats/Chats.js

```diff
@@ -1,5 +1,85 @@
+import { useRef } from 'react' // add useState
+import { useSelector } from 'react-redux' // add useDispatch
+import { sortArrayOfObjects } from '../../utils/helper'
+import StartAChatButton from '../button/StartAChatButton'
+import ModalStartAChat from '../modal/ModalStartAChat'
+import LatestChats from './LatestChats'
+
+// temporary mock data that will be replaced with data from chats store
+import { chatsMockData } from './mock_messages'
+
 function Chats() {
-  return <div>Chats Component</div>
+  // ******todo******
+  // import action from chat store and dispatch an action that sets the current
+  // friend id with an id of a friend that has been clicked
+  // ****************
+
+  // Create the state variable for the selected friend to start a chat
+  // const [selectedFriendId, setSelectededFriendId] = useState()
+
+  // Get the reducer dispatch action
+  // const dispatch = useDispatch()
+
+  // Create handler for selecting the friend when clicking either last message or a friend
+  // const handleSelectFriend = (selectededFriendId) => {
+  //   setSelectedFriendId(selectededFriendId) - set the selected friend id
+  //   dispatch(currentFriend(selectedFriendId) - dispatch an action
+  //   modalRef.current.open && onCloseModal() - close the modal if opened
+  // }
+
+  // Get friends list from the user state slice
+  const friends = useSelector(({ user }) => user.friends)
+  // Get the modal ref
+  const modalRef = useRef()
+  // Get the list of the latest messages, one message per contact
+  const latestChats = chatsMockData.reduce(
+    (newArr, currentChat) => {
+      //  Use provided friend id (from user messages) to get the friend from the friends data
+      const friendId = currentChat.id
```

Comment:
Just a reminder here that a chat  has a friendId field and a messages field

Reply:
In this reduce function we're iterating over `Chats` array.
In each iteration of the `Chats` array we have `currentChat` object that contains friendId field (`currentChat.id`) and messages field (`currentChat.messages`). I attached the friend field to the `friendId`. Perhaps I could've just leave it as `currentChat.id`, but I thought it would be clearer.
Or did you mean something else?

Reply:
I meant that the chats array returned from the backend contains chat objects that have two fields: friendId and messages